### PR TITLE
Refactor generation "where"

### PR DESCRIPTION
**Pull request by justprogrammer2226**
**Related issue: #204**

### CODE REVIEW

File: Cooper.ORM/DbTools.cs [hunk 1]
```diff
@@ -48,89 +19,191 @@ public static string GetVariableAttribute(string attribute)
...
+    }
 
-            return String.Format("DELETE FROM {0} {1}", table, where);
+    public class WhereRequest
+    {
+        private StringBuilder whereRequest = new StringBuilder();
+
+        private static readonly string[] operators = new string[] { "=", ">=", "<=", ">", "<", "IS NULL", "IS NOT NULL", "IN" };
+
+        private static string GetOperatorstring(Operators requestOperator)
```
Comment (Hunroll) [inline Cooper.ORM/DbTools.cs]: GetOperatorString

File: Cooper.ORM/DbTools.cs [hunk 2]
```diff
@@ -48,89 +19,191 @@ public static string GetVariableAttribute(string attribute)
...
-        public static string CreateDeleteQuery(string table, WhereRequest[] whereRequests = null)
+        public static string Wrapstring(string value)
         {
-            string where = CreateWhereRequest(whereRequests);
+            return string.Format("\'{0}\'", value);
+        }
+    }
 
-            return String.Format("DELETE FROM {0} {1}", table, where);
+    public class WhereRequest
```
Comment (Hunroll) [inline Cooper.ORM/DbTools.cs]: May be it should be in another file?

File: Cooper.ORM/DbTools.cs [hunk 3]
```diff
@@ -48,89 +19,191 @@ public static string GetVariableAttribute(string attribute)
...
-                where = " WHERE";
-                foreach (WhereRequest request in whereRequests)
+                if (op == Operators.In)
+                {
+                    whereRequest.Append(" IN ");
+                    whereRequest.Append("(");
+                    whereRequest.Append(string.Join(", ", attributeValues));
+                    whereRequest.Append(")");
+                }
+                else if (attributeValues.Length == 1)
```
Comment (Hunroll) [inline Cooper.ORM/DbTools.cs]: not needed case. Let it go by main if. So it will be attr IN (3), not a problem

File: Cooper.ORM/DbTools.cs (hunk 2 above)
Comment (justprogrammer2226) [reply Cooper.ORM/DbTools.cs]: Yes, you're right, I’ll fix it now.

File: Cooper.ORM/DbTools.cs [hunk 4]
```diff
@@ -48,89 +19,191 @@ public static string GetVariableAttribute(string attribute)
...
+            {
+                whereRequest.Append($" {GetOperatorstring(op)}");
+            }
+            else if (attributeValues.Length > 0)
             {
-                where = " WHERE";
-                foreach (WhereRequest request in whereRequests)
+                if (op == Operators.In)
+                {
+                    whereRequest.Append(" IN ");
```
Comment (Hunroll) [inline Cooper.ORM/DbTools.cs]: Make it in one append ("IN (" + string.Join(...)+")"

File: Cooper.ORM/DbTools.cs [hunk 5]
```diff
@@ -48,89 +19,191 @@ public static string GetVariableAttribute(string attribute)
...
+                throw new ArgumentException("For all operators (except NULL and NOT NULL), the attributeValues array must be initialized with at least one value. " +
+                    "For all operators (except NULL, NOT NULL, IN), the attributeValues array must be initialized with one value.");
+            }
+        }
 
-            return where;
+        public WhereRequest(WhereRequest where)
+        {
+            whereRequest.Append("(");
+            whereRequest.Append(where.ToString().Replace("WHERE ", ""));
```
Comment (Hunroll) [inline Cooper.ORM/DbTools.cs]: Big problem. If I we have search by login and I have login "MrNOWHERE hello". You know what will be)

**Review by Hunroll — CHANGES_REQUESTED** [review]
.Where, .And, .Or, a lot of code duplication. One method should be used in all places to generate all possible statements.

File: Cooper.ORM/DbTools.cs (hunk 4 above)
Comment (justprogrammer2226) [reply Cooper.ORM/DbTools.cs]: Got it.

File: Cooper.ORM/DbTools.cs (hunk 1 above)
Comment (justprogrammer2226) [reply Cooper.ORM/DbTools.cs]: I need to be careful with ctrl+h:)

File: Cooper.ORM/DbTools.cs (hunk 3 above)
Comment (justprogrammer2226) [reply Cooper.ORM/DbTools.cs]: You're right. I'll change it.

**Review by Hunroll — APPROVED** [review]

**Pull request merged**